4-WebApp-your-API/4-1-MyOrg/README-use-keyvault-certificate.md

```diff
@@ -0,0 +1,77 @@
+
+### About Managed Identities for Azure Resources
+
+Azure Key Vault provides a way to securely store credentials, secrets, and other keys, but your code has to authenticate to Key Vault to retrieve them. The [managed identities for Azure resources](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) feature in Azure Active Directory (Azure AD) solves this problem. The feature provides Azure services with an automatically managed identity in Azure AD. You can use the identity to authenticate to any service that supports Azure AD authentication, including Key Vault, without any credentials in your code.
+
+#### **Managed Identities for Azure Resources**
+
+In a daemon application scenario, Managed Identity will work if you have it deployed it in an [Azure Virtual Machine](https://azure.microsoft.com/services/virtual-machines/) or [Azure Web Job](https://docs.microsoft.com/en-us/azure/app-service/webjobs-create). Please, read [this documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) to understand how Managed Identity works with an Azure VM.
+
+##### Configure Managed identity on Azure VM to access Key Vault
+
+To authenticate to Key Vault using your Azure VM, you must first grant it permissions to Key Vault using the **Key Vault Access Policies**. To do that, follow the steps:
+
+1. On Azure Portal, note the name of the Azure VM where you deployed the daemon application.
+1. [Enable managed identity on the virtual machine](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/qs-configure-portal-windows-vm).
+1. On Azure Portal, navigate to **Key Vaults** and select the one that you want the daemon application's VM to access.
+1. Then click on **Access policies** menu and click on **+Add Access Policy**.
+1. Select an adequate template from the dropdown "Configure from template" (ie "Secret & Certificate Management") or set the permissions manually (this sample requires the permission **GET** for Secret and Certificate to be checked).
+1. For **Select principal**, search for the Azure VM *name* or *ObjectId*, select it and click on **Select** button.
+1. Click on **Add**.
+1. Then, **Save**.
+
+For more information about Key Vault, take a look at these links:
+
+- [Key Vault documentation](https://docs.microsoft.com/en-us/azure/key-vault/)
+- [Managed Identity Key Vault sample for dotnet](https://github.com/Azure-Samples/app-service-msi-keyvault-dotnet)
+
+For more information about AzureVM and Managed Identities for Azure Resources, take a look at these links:
+
+- [Managed Identity documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview)
+- [AzureVM documentation](https://azure.microsoft.com/en-us/services/virtual-machines/)
```

Comment:
ahem, lets stay clear from all VM matters, its a different world altogether and we dont wanna go there

Reply:
removed, replaced by general links